packages/connect-form/src/components/advanced-options-tabs/general/schema-input.tsx

```diff
@@ -0,0 +1,169 @@
+import { css } from '@emotion/css';
+import React from 'react';
+import {
+  Banner,
+  BannerVariant,
+  Description,
+  Label,
+  RadioBox,
+  RadioBoxGroup,
+  spacing,
+} from '@mongodb-js/compass-components';
+import ConnectionStringUrl from 'mongodb-connection-string-url';
+
+import { defaultHostname } from '../../../constants/default-connection';
+import { SetConnectionField } from '../../../hooks/use-connect-form';
+
+enum MONGODB_SCHEMA {
+  MONGODB = 'MONGODB',
+  MONGODB_SRV = 'MONGODB_SRV',
+}
+
+const descriptionStyles = css({
+  marginTop: spacing[1],
+});
+
+const regularSchemaDescription =
+  'Standard Connection String Format. The standard format of the MongoDB connection URI is used to connect to a MongoDB deployment: standalone, replica set, or a sharded cluster.';
+const srvSchemaDescription =
+  'DNS Seed List Connection Format. The +srv indicates to the client that the hostname that follows corresponds to a DNS SRV record.';
+
+function updateConnectionStringToStandard(
+  connectionStringUrl: ConnectionStringUrl
+): ConnectionStringUrl {
+  if (!connectionStringUrl.isSRV) {
+    // Already is standard schema, nothing to do.
+    return connectionStringUrl;
+  }
+
+  const newConnectionString = connectionStringUrl.toString();
+
+  const newConnectionStringUrl = new ConnectionStringUrl(
+    newConnectionString.replace('mongodb+srv://', 'mongodb://')
+  );
+
+  newConnectionStringUrl.hosts = [`${newConnectionStringUrl.hosts[0]}:27017`];
+
+  return newConnectionStringUrl;
+}
+
+function updateConnectionStringToSRV(
+  connectionStringUrl: ConnectionStringUrl
+): ConnectionStringUrl {
+  if (connectionStringUrl.isSRV) {
+    // Already is srv schema, nothing to do.
+    return connectionStringUrl;
+  }
+
+  let newConnectionStringUrl = connectionStringUrl.clone();
+
+  // Only include one host without port.
+  const newHost =
+    newConnectionStringUrl.hosts.length > 0
+      ? newConnectionStringUrl.hosts[0].substring(
+          0,
+          newConnectionStringUrl.hosts[0].indexOf(':') === -1
+            ? undefined
+            : newConnectionStringUrl.hosts[0].indexOf(':')
+        )
+      : defaultHostname;
+  newConnectionStringUrl.hosts = [newHost];
+
+  const newConnectionString = newConnectionStringUrl.toString();
+  newConnectionStringUrl = new ConnectionStringUrl(
+    newConnectionString.replace('mongodb://', 'mongodb+srv://')
+  );
+
+  // SRV connections can't have directConnection set.
+  if (newConnectionStringUrl.searchParams.get('directConnection')) {
+    newConnectionStringUrl.searchParams.delete('directConnection');
```

Comment:
Are you thinking we shouldn't unset `directConnection` when switching to srv? We could instead show a warning that it's set when the user switches. 

We'll definitely want to show a warning if a user pastes a connection string with `srv+mongodb` and `directConnection=true`, but since it's a valid connection string we allow them to try to connect. We'll be doing some of the connecting error and warning states in later work.

Two updates to hopefully help this:
Updated the direct connection input to always show if it's set to true (so if a user pastes a connection string with `srv+mongodb` and `directConnection=true` we'll show the checkbox so they can unset it and hide it when they unset it.
Updated switching to standard schema to automatically set `directConnection=true`.

Reply:
I see what you mean, probably if we don't "unset" the direct connection it would be a bit weird when we switch from MongoDB to mongodb+srv. So yeah maybe unsetting that field is necessary. 

`directConnection=true` though is a workaround for SSH tunnels though that only a few people should use, and with the SOCKS5 support should not be needed at all. I would not set that by default.

In my opinion, we should aim not to change the connection string at all if possible with defaults, after all that approach already proved to cause more harm than good both for us and for users, and has other disadvantages in the long run, since we need to make assumptions that may not be true over time. 

For example `mongodb://secondary:27019` is a legit seed list for a replica set, and would work with a driver and connect to the primary node, allowing writes.

If we add `mongodb://secondary:27019?directConnection=true` we would forge a different connection string that is going to connect to a secondary and probably not work as expected.

To fix the issue with switching from non-SRV to SRV   ..  what about keeping some 'volatile' state instead? 

By default we don't set or change anything.

When we switch to SRV we save the 'old non-SRV options' and if we switch back we reload them, this would also apply to hosts. If I have 3 hosts and I switch to SRV, only the first one is used but the other 2 are kept in memory in case the user would go back to that.

We do need to handle the state outside of the fields though, these fields should be probably not be dealing with the connection string directly after all.

Reply:
Reverted that defaulting of `directConnection=true` when switching back 👍 
Agreed, the simpler connection string the better.

I think if we want to persist the various hosts when switching non-srv to srv and back we might want to do it in a separate pr. It's a bit of an extra feature work since I think we'll need to pass the current connection id down to the component to know to drop the old state when the connection id changes. Does that sound ok? I'm cool with doing it here too, but it can be done in isolation since its a benefit to this.

Reply:
Sure!